appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/HealthCheckService.java

```diff
@@ -115,6 +125,8 @@
     private final Map<String, ClassLoader> applicationClassLoaders = new ConcurrentHashMap<>();
     private final List<String> applicationsLoaded = new CopyOnWriteArrayList<>();
 
+    private final AtomicReference<Map<String, Set<String>>> checksCollected = new AtomicReference<>();
```

Comment:
FYI: This state is needed to know when collecting watches which metrics have been collected before.